compiler/src/test/java/com/squareup/anvil/compiler/testing/AnvilAnnotationsTest.kt

```diff
@@ -0,0 +1,56 @@
+package com.squareup.anvil.compiler.testing
+
+import com.rickbusarow.kase.DefaultTestEnvironment
+import com.rickbusarow.kase.Kase1
+import com.rickbusarow.kase.KaseTestFactory
+import com.rickbusarow.kase.ParamTestEnvironmentFactory
+import com.rickbusarow.kase.files.HasWorkingDir
+import com.rickbusarow.kase.files.TestLocation
+import com.rickbusarow.kase.kases
+import com.squareup.anvil.compiler.isFullTestRun
+import kotlin.reflect.KClass
+
+abstract class AnvilAnnotationsTest(
+  firstAnnotation: KClass<out Annotation>,
+  vararg fullTestRunAnnotations: KClass<out Annotation>,
+) : KaseTestFactory<
```

Comment:
Is there a way to use `KaseTestFactory` via composition or a test rule instead of introducing a base class?

Reply:
No to the Rule, technically yes to the base class.  But the intent is for it to be used for the other tests with annotation parameters as well (there are five `Binding___Test` in total).  The setup for each is identical except for those parameters.